x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java

```diff
@@ -586,6 +596,19 @@ public AutoscalingDeciderResult scale(Settings configuration, AutoscalingDecider
                     .build()));
     }
 
+    static AutoscalingCapacity ensureScaleDown(AutoscalingCapacity scaleDownResult, AutoscalingCapacity currentCapacity) {
```

Comment:
The fact that this method is needed suggests there are bugs in the other calculations.  It's good to have it as a safety net, but I think it would be good if it could log a warning if it changes the answer by more than the few bytes that could be attributed to rounding errors.

Reply:
> but I think it would be good if it could log a warning if it changes the answer by more than the few bytes that could be attributed to rounding errors.

100%, I will add a log line.

I do think we need this method as this has been an issue in a couple known scenarios and it can be exceptionally frustrating to troubleshoot.